api/src/ginger_calls.rs

```diff
@@ -1,85 +1,117 @@
 }, merkle_tree::field_based_mht::{
-    smt::{BigMerkleTree, LazyBigMerkleTree, Coord, OperationLeaf},
+    // smt::{BigMerkleTree, LazyBigMerkleTree, Coord, OperationLeaf},
     optimized::FieldBasedOptimizedMHT,
-    poseidon::{MNT4753_PHANTOM_MERKLE_ROOT as PHANTOM_MERKLE_ROOT, MNT4753_MHT_POSEIDON_PARAMETERS as MHT_PARAMETERS},
+    parameters::tweedle_fr::TWEEDLE_MHT_POSEIDON_PARAMETERS as MHT_PARAMETERS,
+    // parameters::mnt4753::{MNT4753_PHANTOM_MERKLE_ROOT as PHANTOM_MERKLE_ROOT, MNT4753_MHT_POSEIDON_PARAMETERS as MHT_PARAMETERS},
     FieldBasedMerkleTree, FieldBasedMerkleTreePrecomputedEmptyConstants,
     FieldBasedMerkleTreeParameters, BatchFieldBasedMerkleTreeParameters,
     FieldBasedMerkleTreePath, FieldBasedBinaryMHTPath,
 }, signature::{
     FieldBasedSignatureScheme, schnorr::field_based_schnorr::{
-        FieldBasedSchnorrSignatureScheme, FieldBasedSchnorrSignature
+        FieldBasedSchnorrSignatureScheme, FieldBasedSchnorrSignature,
+        FieldBasedSchnorrPk,
     },
-}, vrf::{FieldBasedVrf, ecvrf::*}, ActionLeaf};
-use proof_systems::groth16::{
-    Proof, create_random_proof,
-    prepare_verifying_key, verify_proof,
-};
+}, vrf::{FieldBasedVrf, ecvrf::*}/*, ActionLeaf*/};
+
+use marlin::*;
+use blake2::Blake2s;
+use poly_commit::ipa_pc::InnerProductArgPC;
+// use proof_systems::groth16::{
+//     create_random_proof,
+//     prepare_verifying_key, verify_proof,
+// };
 use demo_circuit::{
     constants::{
         VRFParams, VRFWindow,
     },
     naive_threshold_sig::*
 };
 use rand::{
-    SeedableRng, rngs::OsRng
+    SeedableRng, rngs::OsRng, thread_rng
 };
 use rand_xorshift::XorShiftRng;
 
 use std::{
-    fs::File, io::Result as IoResult, path::Path
+    fs::File, io::Result as IoResult/*, path::Path*/
 };
 use lazy_static::*;
 
 pub type FieldElement = Fr;
 
-pub const FIELD_SIZE: usize = 96; //Field size in bytes
-pub const SCALAR_FIELD_SIZE: usize = FIELD_SIZE;// 96
+// #[derive(Clone)]
+// struct MarlinNoLCNoZk;
+//
+// impl MarlinConfig for MarlinNoLCNoZk {
+//     const LC_OPT: bool = false;
+//     const ZK: bool = false;
+// }
+
+type IPAPC = InnerProductArgPC<PCAffine, Blake2s>;
+// type MarlinInst = Marlin<Fr, IPAPC, Blake2s, MarlinNoLCNoZk>;
+type MarlinInst = Marlin<Fr, IPAPC, Blake2s>;
+
+pub const FIELD_SIZE: usize = 32; //Field size in bytes
+pub const SCALAR_FIELD_SIZE: usize = FIELD_SIZE;// 32
 pub const G1_SIZE: usize = 193;
 pub const G2_SIZE: usize = 385;
 
 pub const SCHNORR_PK_SIZE: usize = G1_SIZE; // 193
-pub const SCHNORR_SK_SIZE: usize = SCALAR_FIELD_SIZE; // 96
+pub const SCHNORR_SK_SIZE: usize = SCALAR_FIELD_SIZE; // 32
 pub const SCHNORR_SIG_SIZE: usize = 2 * FIELD_SIZE; // 192
 
 pub const VRF_PK_SIZE: usize = G1_SIZE; // 193
-pub const VRF_SK_SIZE: usize = SCALAR_FIELD_SIZE; // 96
+pub const VRF_SK_SIZE: usize = SCALAR_FIELD_SIZE; // 32
 pub const VRF_PROOF_SIZE: usize = G1_SIZE + 2 * FIELD_SIZE; // 192
 
 pub const ZK_PROOF_SIZE: usize = 2 * G1_SIZE + G2_SIZE;  // 771
```

Comment:
This is the size of a Groth proof, we should change to the size of a standard Marlin one

Reply:
A quick note on this, @i-Alex please I would like your feedback too.
The size of a Marlin proof may change according to the specific circuit for which the proof is created and according to some parameters that should be declared at SC setup time.
My proposal is that, to know the serialized size of a proof, we can expose an additional function get_sc_proof_size() to be called before serializing, and SDK size this value can be saved somewhere (it won't change for the single SC, the problem is that it may change from SC to SC).

Reply:
(If needed)

Reply:
Variable "ZK_PROOF_SIZE" removed, and size of proof now is not fixed